UWP/UWPHelpers/StorageManager.cpp

```diff
@@ -0,0 +1,331 @@
+}
+std::string GetDocumentsFolder() {
+	// Requires 'documentsLibrary' capability
+	return FromPlatformString(KnownFolders::DocumentsLibrary->Path);
+}
+std::string GetMusicFolder() {
+	// Requires 'musicLibrary' capability
+	return FromPlatformString(KnownFolders::MusicLibrary->Path);
+}
+std::string GetPreviewPath(std::string path) {
+	std::string pathView = path;
+
+	pathView = ReplaceAll(pathView, "/", "\\");
+	pathView = ReplaceAll(pathView, GetLocalFolder(), "LocalState");
+	pathView = ReplaceAll(pathView, GetTempFolder(), "TempState");
+	pathView = ReplaceAll(pathView, GetInstallationFolder(), "Installation folder");
+	return pathView;
+}
+bool isLocalState(std::string path) {
+	return !_stricmp(GetPreviewPath(path).c_str(), "LocalState");
+}
+#pragma endregion
+
+#pragma region Internal
+Path PathResolver(Path path) {
+	auto root = path.GetDirectory();
+	auto newPath = path.ToString();
+	if (path.IsRoot() || !_stricmp(root.c_str(), "/") || !_stricmp(root.c_str(), "\\")) {
+		// System requesting file from app data
+		newPath = ReplaceAll(newPath, "/", (GetLocalFolder() + (path.size() > 1 ? "/" : "")));
+	}
+	return Path(newPath);
+}
+Path PathResolver(std::string path) {
+	return PathResolver(Path(path));
+}
+
+std::string ResolvePathUWP(std::string path) {
+	return PathResolver(path).ToString();
+}
+#pragma endregion
+
+#pragma region Functions
+std::map<std::string, bool> accessState;
+bool CheckDriveAccess(std::string driveName) {
+	bool state = false;
+
+	HANDLE searchResults;
+	WIN32_FIND_DATA findDataResult;
+	auto keyIter = accessState.find(driveName);
+	if (keyIter != accessState.end()) {
+		state = keyIter->second;
+	}
+	else {
+		try {
+			wchar_t* filteredPath = _wcsdup(ConvertUTF8ToWString(driveName).c_str());
+			wcscat_s(filteredPath, sizeof(L"\\*.*"), L"\\*.*");
+
+			searchResults = FindFirstFileExFromAppW(
+				filteredPath, FindExInfoBasic, &findDataResult,
+				FindExSearchNameMatch, NULL, 0);
+
+			state = searchResults != NULL && searchResults != INVALID_HANDLE_VALUE;
+			if (state) {
+				FindClose(searchResults);
+			}
+			// Cache the state
+			accessState.insert(std::make_pair(driveName, state));
+		}
+		catch (...) {
+		}
+	}
+	if (!state) {
+		state = IsRootForAccessibleItems(driveName);
+	}
+	return state;
+}
+
+bool isWriteMode(const char* mode) {
+	return (!strcmp(mode, "w") || !strcmp(mode, "wb") || !strcmp(mode, "wt") || !strcmp(mode, "at") || !strcmp(mode, "a"));
+}
+bool isAppendMode(const char* mode) {
+	return (!strcmp(mode, "at") || !strcmp(mode, "a"));
+}
+
+FILE* GetFileStreamFromApp(std::string path, const char* mode) {
+
+	FILE* file{};
+
+	auto pathResolved = Path(ResolvePathUWP(path));
+	HANDLE handle;
+	auto access = GENERIC_READ;
+	auto share = FILE_SHARE_READ;
+	auto creation = OPEN_EXISTING;
+	bool isWrite = isWriteMode(mode);
+	bool isAppend = isAppendMode(mode);
+
+	if (isWrite) {
+		access = GENERIC_WRITE;
+		share = FILE_SHARE_WRITE;
+		creation = isAppend ? OPEN_ALWAYS : CREATE_ALWAYS;
```

Comment:
Actually, `"a"` means that all writes should be forced to the end of the file.

If the file was opened for "r+" (or rb+, or r+b, or rt+, or r+t), it would be writable but would not be created if it doesn't exist.  "w+" would create still.

-[Unknown]

Reply:
After the commit below it should be more accurate, I didn't know there are too many cases, hopefully all covered in the new function